crates/bevy_state/src/state/freely_mutable_state.rs

```diff
@@ -17,27 +17,33 @@ use super::{take_next_state, transitions::*};
 pub trait FreelyMutableState: States {
     /// This function registers all the necessary systems to apply state changes and run transition schedules
     fn register_state(schedule: &mut Schedule) {
+        schedule.configure_sets((
+            ApplyStateTransition::<Self>::default()
+                .in_set(StateTransitionSteps::DependentTransitions),
```

Comment:
Why the change from `RootTransitions` to `DependentTransitions` for `FreelyMutableState` types?

Reply:
Copy paste "error", this part of code is HEAVILY shared between all state traits, the only difference being the amount of dependencies (root states = none, sub/comp one [manual impl] or more [macros])

Reply:
I'll keep this and remove the `RootTransitions` instead